Formula/openssh.rb

```diff
@@ -71,7 +71,10 @@ def install
 
       # FIXME: `ssh-keygen` errors out when this is built with optimisation.
       # Reported upstream at https://bugzilla.mindrot.org/show_bug.cgi?id=3584
-      ENV.O0 if Hardware::CPU.intel? && MacOS.version == :ventura && version == Version.new("9.3p1")
+      # Also can segfault at runtime: https://github.com/Homebrew/homebrew-core/issues/135200
+      if Hardware::CPU.intel? && DevelopmentTools.clang_build_version == 1403
+        inreplace "configure", "-fzero-call-used-regs=all", "-fzero-call-used-regs=used"
+      end
```

Comment:
Corresponding `configure.ac`:
```
	# clang 15 seems to have a bug in -fzero-call-used-regs=all.  See
	# https://bugzilla.mindrot.org/show_bug.cgi?id=3475 and
	# https://github.com/llvm/llvm-project/issues/59242
	case "$CLANG_VER" in
	15.*) OSSH_CHECK_CFLAG_COMPILE([-fzero-call-used-regs=used]) ;;
	*)    OSSH_CHECK_CFLAG_COMPILE([-fzero-call-used-regs=all]) ;;
	esac
```

Apple clang 14.0.3 is based on LLVM 15. Upstream fix would require changing `configure.ac` logic to detect this particular version. Just using simple inreplace for now and maybe we can add details to existing upstream bug.

Haven't checked if root cause of every reported OpenSSH issue, but worked for me locally to fix segfault issue.

Corresponding LLVM bug fix https://reviews.llvm.org/D139679 went in 15.0.7. Looks like it was zeroing out some registers which could lead to original `ssh-keygen` issue (i.e. argc/argv getting zeroed) and the segfault issue where some variables were NULL.